aggregators/aggregator_test.go

```diff
@@ -1123,11 +1118,80 @@ func BenchmarkAggregateBatch(b *testing.B) {
 			},
 		},
 	}
+	b.ResetTimer()
+
 	for i := 0; i < b.N; i++ {
 		if err := agg.AggregateBatch(context.Background(), "test", batch); err != nil {
 			b.Fatal(err)
 		}
 	}
+
+	if agg.batch != nil {
+		if err := agg.batch.Commit(pebble.Sync); err != nil {
+			b.Fatal(err)
+		}
+		if err := agg.batch.Close(); err != nil {
+			b.Fatal(err)
+		}
+		agg.batch = nil
+	}
+	if err := agg.db.Close(); err != nil {
+		b.Fatal(err)
+	}
+}
+
+func BenchmarkAggregateBatchParallel(b *testing.B) {
+	b.ReportAllocs()
+	agg, err := New(
+		WithDataDir(b.TempDir()),
+		WithLimits(Limits{
+			MaxSpanGroups:                         1000,
+			MaxSpanGroupsPerService:               100,
+			MaxTransactionGroups:                  1000,
+			MaxTransactionGroupsPerService:        100,
+			MaxServiceTransactionGroups:           1000,
+			MaxServiceTransactionGroupsPerService: 100,
+			MaxServices:                           100,
+			MaxServiceInstanceGroupsPerService:    100,
+		}),
+		WithProcessor(noOpProcessor()),
+		WithAggregationIntervals([]time.Duration{time.Second, time.Minute, time.Hour}),
+	)
+	if err != nil {
+		b.Fatal(err)
+	}
```

Comment:
Maybe encapsulate this + cleanup in a helper?

```go
func newBenchAggregator(b *testing.B) *Aggregator {
	agg, err := New(
		WithDataDir(b.TempDir()),
		WithLimits(Limits{
		MaxSpanGroups:                         1000,
			MaxSpanGroupsPerService:               100,
			MaxTransactionGroups:                  1000,
			MaxTransactionGroupsPerService:        100,
			MaxServiceTransactionGroups:           1000,
			MaxServiceTransactionGroupsPerService: 100,
			MaxServices:                           100,
			MaxServiceInstanceGroupsPerService:    100,
		}),
		WithProcessor(noOpProcessor()),
		WithAggregationIntervals([]time.Duration{time.Second, time.Minute, time.Hour}),
	)
	if err != nil {
		b.Fatal(err)
	}
	b.Cleanup(func() {
		if agg.batch != nil {
			if err := agg.batch.Commit(pebble.Sync); err != nil {
				b.Fatal(err)
			}
			if err := agg.batch.Close(); err != nil {
				b.Fatal(err)
			}
			agg.batch = nil
		}
		if err := agg.db.Close(); err != nil {
			b.Fatal(err)
		}
	})
	return agg
}
```

Reply:
Or you could use `testing.TB` if it's appropriate to have the same parameters for both unit tests & benchmarks.